pkg/tokens/builder/builder.go

```diff
@@ -0,0 +1,120 @@
+package builder
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/status-im/go-wallet-sdk/pkg/common"
+	"github.com/status-im/go-wallet-sdk/pkg/tokens/parsers"
+	"github.com/status-im/go-wallet-sdk/pkg/tokens/types"
+)
+
+const (
+	NativeTokenListID = "native"
+
+	EthereumNativeCrossChainID = "eth-native"
+	EthereumNativeSymbol       = "ETH"
+	EthereumNativeName         = "Ethereum"
+
+	BinanceSmartChainNativeCrossChainID = "bsc-native"
+	BinanceSmartChainNativeSymbol       = "BNB"
+	BinanceSmartChainNativeName         = "BNB"
+)
+
+var (
+	ErrEmptyRawTokenList = fmt.Errorf("raw token list data is empty")
+	ErrParserIsNil       = fmt.Errorf("parser is nil")
+)
+
+// Builder builds token lists into a single list of unique tokens.
+type Builder struct {
+	chains     []uint64
+	tokens     map[string]*types.Token
+	tokenLists map[string]*types.TokenList
+}
+
+// New creates a new Builder instance.
+func New(chains []uint64) *Builder {
+	return &Builder{
+		chains:     chains,
+		tokens:     make(map[string]*types.Token),
+		tokenLists: make(map[string]*types.TokenList),
+	}
+}
+
+// GetTokens returns the list of unique tokens of all added token lists.
+func (b *Builder) GetTokens() map[string]*types.Token {
+	return b.tokens
+}
+
+// GetTokenLists returns the list of added token lists.
+func (b *Builder) GetTokenLists() map[string]*types.TokenList {
+	return b.tokenLists
+}
+
+func getNativeToken(chainID uint64) *types.Token {
```

Comment:
it seems very random to be adding native tokens for these two specific chains here. If Status needs these two tokens at startup for whatever reason, it should insert the appropriate list using the exposed API

Reply:
That's part of `AddNativeTokenList` function, so if the consumer doesn't need native tokens then just doesn't need to call that function. If needs different tokens can use `AddTokenList`.